trust_stores_observatory/store_fetcher/java_fetcher.py

```diff
@@ -18,170 +20,161 @@
 
-    def fetch(self,
-              cert_repo: RootCertificatesRepository,
-              should_update_repo: bool = True) -> TrustStore:
-        path_to_security = '/lib/security/'
-        cacerts = 'cacerts'
-        blacklisted = 'blacklisted.certs'
-        default_password = 'changeit'  # default password for key store
+    def fetch(
+            self,
+            cert_repo: RootCertificatesRepository,
+            should_update_repo: bool=True
+    ) -> TrustStore:
+        # Fetch the latest JRE package
+        final_url = self._get_latest_download_url()
+        request = Request(
+            final_url,
+            # Cookie set when 'Accept License Agreement' is selected
+            headers={'Cookie': 'oraclelicense=accept-securebackup-cookie'}
+        )
+        response = urlopen(request)
 
+        # Parse the JRE package
+        jre_temp_file = NamedTemporaryFile(delete=False)
         try:
-            cookie_header = {}
-            # cookie set when 'Accept License Agreement' is selected
-            cookie_header[
-                'Cookie'] = 'oraclelicense=accept-securebackup-cookie'
-            url = self._get_latest_download_url()
-            req = Request(url, headers=cookie_header)
-
-            download_content = urlopen(req)
-
-            with NamedTemporaryFile(mode='wb') as fh:
-                fh.write(download_content.read())
-                with tarfile.open(name=fh.name, mode='r:gz') as tar_file:
-                    cacert_filename = [
-                        i for i in tar_file.getnames()
-                        if path_to_security + cacerts in i
-                    ]
-                    blacklist_filename = [
-                        i for i in tar_file.getnames()
-                        if path_to_security + blacklisted in i
-                    ]
-                    try:
-                        s_cacert_filename = cacert_filename[0]
-                        s_blacklist_filename = blacklist_filename[0]
-                        version = s_cacert_filename[:s_cacert_filename.find('/')]
-                    except Exception as e:
-                        raise e
-
-                    blacklist_cert_file: Any = tar_file.extractfile(s_blacklist_filename)
-                    cacert_file: Any = tar_file.extractfile(s_cacert_filename)
-                    with NamedTemporaryFile(mode='wb') as blacklist:
-                        blacklist.write(blacklist_cert_file.read())
-                        blacklist.flush()
-                        with open(blacklist.name) as blacklisted_file:
-                            blacklisted_certs = blacklisted_file.read()
-
-                    with NamedTemporaryFile(mode='wb') as fh2:
-                        fh2.write(cacert_file.read())
-                        fh2.flush()
-                        key_store = jks.KeyStore.load(fh2.name, default_password)
-
-        except Exception:
-            raise ValueError('Could not fetch file')
```

Comment:
Better to just let the initial exception propagate instead of raising a more generic exception.